src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -66,9 +87,9 @@ export function usePanZoom(camera: OrthographicCamera): ReactThreeFiber.Events {
       const { x: pointX, y: pointY } = projectedPoint;
       const { x: startX, y: startY } = startOffsetPosition.current;
 
-      camera.position.set(startX - pointX, startY - pointY, camera.position.z);
+      moveCamera(startX - pointX, startY - pointY);
     },
-    [camera, startOffsetPosition]
+    [camera, moveCamera]
   );
 
   const onWheel = useCallback(
```

Comment:
Just a remark: when I pan while zoomed and then zoom out, it is possible to be out of bounds. 
A possibility to fix this would be to add `moveCamera(camera.position.x, camera.position.y)` at the end of `onWheel` to go through the clamping code. But there is perhaps a better solution...

Reply:
That's the p(l)an! ;)

Reply:
I'll deal with moving the camera during zoom in the next PR, along with centring the zoom on the cursor.